src/ray/core_worker/core_worker.cc

```diff
@@ -15,7 +16,10 @@ CoreWorker::CoreWorker(
       raylet_socket_(raylet_socket),
       log_dir_(log_dir),
       worker_context_(worker_type, job_id),
-      io_work_(io_service_) {
+      io_work_(io_service_),
+      heartbeat_timer_(io_service_,
```

Comment:
Can you suppress redundant updates here?

Reply:
Done.

Reply:
Or is your suggestion to suppress updates when the entire set of active IDs is unchanged? That should work and would be simple enough but it's unclear to me if it would have a significant performance impact.

Reply:
It's only a few lines of code and clear enough so I am suppressing them now.